File: evals/code-intelligence/fixtures/batch-d/cpp/src/item_service.cpp
#include "../include/item_service.hpp"

namespace demo {
Item ItemService::lookup(const std::string &id) const { return Item{id}; }
Item ItemService::find(const std::string &id) const { return Item{id}; }
Item ItemService::find(long id) const { return Item{std::to_string(id)}; }

Item load_item(const ItemService &service, const std::string &id) {
  return service.lookup(id);
}

Item ambiguous(const std::string &id) { return find(id); }
}

File: evals/code-intelligence/fixtures/batch-e/ruby/lib/demo/item_service.rb
require_relative "item"

module Demo
  module Logging
    def log_item(id)
      id
    end
  end

  class BaseService
  end

  class ItemService < BaseService
    include Logging

    def find(id)
      Item.new(id)
    end

    def summary(item)
      item.id
    end
  end

  def self.describe(item)
    service = ItemService.new
    service.summary(item)
  end
end
